dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CommunityCollectionLinkRepository.java

```diff
@@ -48,10 +60,25 @@
             if (community == null) {
                 throw new ResourceNotFoundException("No such community: " + communityId);
             }
-            List<Collection> collections = community.getCollections();
-            return converter.toRestPage(utils.getPage(collections, optionalPageable), projection);
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
+            Pageable pageable = utils.getPageable(optionalPageable);
+            List<Collection> collections = new LinkedList<Collection>();
+            IndexObjectFactoryFactory indexObjectFactory = IndexObjectFactoryFactory.getInstance();
+            IndexableObject scopeObject = indexObjectFactory.getIndexableObjects(context, community).get(0);
+            DiscoverQuery discoverQuery = new DiscoverQuery();
```

Comment:
Again, it'd be useful to add an inline comment to describe the query we are building.  This looks to be searching for all Collections  (that the user has access to) under a specific Community?

Reply:
done here https://github.com/DSpace/DSpace/pull/2707/commits/94706cfcc28e0d7a9b0d8affa1a4ad468b776e95#diff-c90f703886a8245f9a9e23f499cfca90R66